scheduled_video_capture/scheduled_video_capture.py

```diff
@@ -0,0 +1,344 @@
+
+        return ""
+    
+def get_quality_perc(resolution):
+    if resolution == 'Original':
+        return 100
+    elif resolution == 'Lower':
+        return 75
+    return 50
+
+#Create a global session manager
+capture_sessions = VideoCaptureSessions()
+
+def save_frame(frame, camera_id, task_uuid, task_settings, task_id):    
+    root_path = task_settings.get("root_path", "./assets/saved_videos")
+    file_type = task_settings.get("file_type", "mp4")
+    resolution = task_settings.get("resolution", "Lowest")
+    interval = int(task_settings.get("interval", 3600))
+    interval_units = task_settings.get("interval_units", "Seconds")
+    if interval_units != "Seconds":
+        interval *= TIME_UNITS.get(interval_units, 1)
+    duration = int(task_settings.get("clip_length", interval))
+    duration_units = task_settings.get("clip_length_units", "Seconds")
+    if duration_units != "Seconds":
+        duration *= TIME_UNITS.get(duration_units, 1)
+    start_time = task_settings.get("start_time", datetime.now(timezone.utc).isoformat())
+    timestamp = datetime.now(timezone.utc).isoformat()
+   
+    #Get frame shape for initialization
+    frame_shape = frame.shape
+   
+    saved_file_path = ""  #Initialize return value
+    
+    #If idling is not set up, initialize it
+    if camera_id not in capture_sessions.idling:
+        capture_sessions.idling[camera_id] = {}
+    if task_uuid not in capture_sessions.idling[camera_id]:
+        capture_sessions.idling[camera_id][task_uuid] = { 'last_interval_position': 0 }
+    
+    should_start = capture_sessions.should_start_capture(camera_id, task_uuid, timestamp, start_time, interval)
+
+    #Check if we have an active session and it should end
+    active_session = camera_id in capture_sessions.sessions and task_uuid in capture_sessions.sessions[camera_id]
+    should_end = active_session and capture_sessions.should_end_session(camera_id, task_uuid, timestamp, duration)
+    
+    if active_session:
+        if should_end or should_start: #End session if duration exceeded or new session needs to start
+            saved_file_path = capture_sessions.end_session(camera_id, task_uuid, root_path, resolution, file_type, task_id, duration)
+        else:
+            capture_sessions.add_frame(camera_id, task_uuid, frame, timestamp)
+    if should_start:
+        capture_sessions.start_session(camera_id, task_uuid, timestamp, frame_shape)
+    
+    return saved_file_path
+
+if __name__ == '__main__':
+    import os
+    import time
+    import cv2
+    
+    #Set environment variable for testing
+    os.environ['CB_SYSTEM_KEY'] = "test_system_key"
+    
+    #Path to test video - replace with your actual path
+    video_path = '/path/to/your/video.mp4'
+    
+    #Initialize camera and task IDs
+    camera_id = "camera_1"
+    task_uuid = "test_task_123"
+    
+    #Create test settings
+    start_time = datetime.now(timezone.utc)
+    task_settings = {
+        "root_path": "/assets/videos",
```

Comment:
nitpick - change the path to `./assets/videos`. The current path try to add it to the root folder.